reactive/kotlinx-coroutines-rx2/test/SchedulerStressTest.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.rx2
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.*
+import org.junit.*
+import java.util.concurrent.*
+
+class SchedulerStressTest : TestBase() {
+    @Before
+    fun setup() {
+        ignoreLostThreads("RxCachedThreadScheduler-", "RxCachedWorkerPoolEvictor-", "RxSchedulerPurge-")
+    }
+
+    /**
+     * Test that we don't get an OOM if we schedule many jobs at once. It's expected that if you don't dispose that you'd
+     * see a OOM error.
+     */
+    @Test
+    fun testSchedulerDisposed(): Unit = runTest {
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+        testRunnableDisposed(scheduler::scheduleDirect)
+    }
+
+    @Test
+    fun testSchedulerWorkerDisposed(): Unit = runTest {
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+        testRunnableDisposed(scheduler.createWorker()::schedule)
+    }
+
+    private suspend fun testRunnableDisposed(block: RxSchedulerBlockNoDelay) {
+        expect(1)
+
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+
+        val n = 2000 * stressTestMultiplier
+        coroutineScope {
+            repeat(n) { i ->
+                launch {
+                    val a = ByteArray(1000000) //1MB
+                    val disposable = block(Runnable {
+                        expectUnreached()
+                        runBlocking {
+                            keepMe(a)
+                        }
```

Comment:
`runBlocking` with `delay` is effectively just a `Thread.sleep`, it would be less code and simpler to just do that here.

Reply:
While less code, it also requires an extra param for the sleep which I think makes it a little dirtier. i'd prefer to keep it as is.